### PR TITLE
 MotorCycle Booking Backend App

### PR DESCRIPTION
**Pull request by Kaiserabbas**
## MotorCycle Booking Backend App

- Created a new Rails app.
- Defined models.
- Generated migration files.
- Set up associations.
- Generated controllers boilerplates.
- Added doctor names to appointments.
- Set up routes.
- Added authentication.
- Implemented a city column in appointments.
- Updated seeds, models, and controllers.
- Added uniqueness validation to username.
- Configured application and CORS initializers.
- Fixed Swagger documentation issues.
- Updated API docs title.
- Implemented token authorization in Swagger.
- Configured production environment settings.
- Added render configuration and script.
- Corrected RSpec and linters errors.

https://github.com/Kaiserabbas/motorcycle_booking_back_end/assets/123939543/928c5788-2655-4da2-9f37-c5d6ef0596d2

https://github.com/Kaiserabbas/motorcycle_booking_front_end/pull/21 to the front-end UI pull request

### CODE REVIEW

File: app/controllers/api/v1/motorcycles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    end
+  end
+
+  def destroy
+    return unless Motorcycle.exists?(params[:id])
+
+    @motorcycle = Motorcycle.destroy(params[:id])
+    authorize! :destroy, @motorcycle
+    render json: { success: true, message: 'Removed Successfully!😁' }, status: :ok
+  end
```
Comment (Meltrust) [inline app/controllers/api/v1/motorcycles_controller.rb]: - [ ] What happens here if the Motorcycle is not successfully destroyed?  Kindly, add some error handling like you did on the create method (line 23). 👍

File: README.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+```
+  rails server
+```
+
+### Run tests <a name="run tests"></a>
+
+To test, run:
+
+```
+  rspec
```
Comment (Meltrust) [inline README.md:104]: - [ ]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!








### Main takeaways:  

- The server and the frontend app on the cloud are working well.  The server code on GitHub does not.
- The cloud live demo was prioritized, which is ok, however, the server Rails app in GitHub, upon cloning the project and trying to make it work, is not actionable because the Cors policy only allows the live demo.
- Upon changing the cors file to allow localhost:3001, the server cannot create a motorcycle using the included [frontend](https://github.com/Kaiserabbas/motorcycle_booking_front_end/pull/21#pullrequestreview-1813845531).
- The [frontend ](https://github.com/Kaiserabbas/motorcycle_booking_front_end/pull/21#pullrequestreview-1813845531)app present on GitHub has a bug that prevents it from starting (easily fixable).
- Most of the review comments are on the frontend repo.

_tip:  You already have a working app in the cloud.  Just make sure the app on GitHub also works well._

#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Comment by Graciano1997** [comment]
Thanks 🙏 very much we will fix it

File: README.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
+<p align="center">
```
Comment (Meltrust) [inline README.md:1]: - [ ] Kindly, go to the frontend repo to see the [review ](https://github.com/Kaiserabbas/motorcycle_booking_front_end/pull/21#pullrequestreview-1814587538)comment. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!










#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Review by Meltrust — APPROVED** [review]
# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight


- Superb job ✔️  



Cheers and Happy coding!👏👏👏


------

**Pull request merged**